cli/src/test/java/com/liferay/blade/cli/BladeTest.java

```diff
@@ -78,15 +78,40 @@ public BladeTest(PrintStream out, PrintStream err, InputStream in, File userHome
 
 	@Override
 	public BladeSettings getBladeSettings() throws IOException {
-		final File settingsFile;
+		File settingsBaseDir;
```

Comment:
This logic looks different from what is in BladeCLI?  That seems weird.  It would make sense that the `BladeTEST` would need to handle blade settings differently.

ALso these are "tests" so we do not need to handle 'migration' or 'backwards compatibility'.  We can setup and control the initial conditions.  WE certainly do need to test the 'migration' use-case which i see you have done in the test class.